File: problem16/p16.c
#include <stdio.h>
#include <stdlib.h>

int power_digit_sum(int num, int pow){

    int i, j, k, carry = 0, size = pow/2, sum = 0;
    int res[size];

    for(i=0; i<size; i++){
        res[i] = 0;
    }

    res[size - 1] = 2;

    for(i=1; i<pow; i++){
        for(j=size-1; j>0; j--){
            res[j] = (res[j] * 2) + carry;
            carry = 0;
            if(res[j] >= 10){
                carry = res[j] / 10;
                res[j] = res[j] % 10;
            }
        }
    }

    for(j=0; j<size; j++){
        sum+= res[j];
    }

    return sum;
    
}

int main()
{
    printf("%d", power_digit_sum(2, 1000));

    return 0;
}

File: problem16/p16.java
import java.math.BigInteger;

public class p16 {

	public static int power_digit_sum(int num, int pow) {
		
		int sum = 0;
		BigInteger bigInt = new BigInteger(Integer.toString(num));
		
		String str = bigInt.pow(pow).toString();
		
		for(int i=0; i<str.length(); i++) {
			sum += Character.getNumericValue(str.charAt(i));
		}
		
		return sum;
		
		
	}
	
	public static void main(String[] args) {
		
		System.out.println(power_digit_sum(2, 1000));
		
	}
	
}
